### PR TITLE
[docs] Update flex parent demo to set min height for overlays

### PR DESCRIPTION
**Pull request by cherniavskii**
Fixes https://github.com/mui/mui-x/issues/14881#issuecomment-2402257931

### CODE REVIEW

**Review by michelengelen — APPROVED** [review]
LGTM :shipit:

File: docs/data/data-grid/layout/FlexGrid.js [hunk 1]
```diff
@@ -26,8 +28,14 @@ export default function FlexGrid() {
           Add a row
         </Button>
       </Stack>
-      <div style={{ display: 'flex', flexDirection: 'column' }}>
-        <DataGrid {...data} rows={data.rows.slice(0, nbRows)} loading={loading} />
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          minHeight: rows.length === 0 ? 200 : undefined,
```
Comment (cherniavskii) [inline docs/data/data-grid/layout/FlexGrid.js:35]: Rationale: https://github.com/mui/mui-x/pull/14901#discussion_r1793890736

**Comment by sampain** [comment]
Would be great if the overlay had a minimum height instead. It's something you can do manually with some CSS.

Ex: 
```
  '& .MuiDataGrid-overlayWrapper': {
    minHeight: '104px',
  },
```

Adding a minimum height on the flex container that you have to calculate based on the size of the overlay that you want seems a bit weird and it's a number that looks arbitrary. If I want a 104px overlay, I need to set a minHeight of 231px. 231px does't look great in the code and needs a comment above it to explain why it's such a specific number. It's also a number you have to change if you add a toolbar to the datagrid for example.

**Comment by cherniavskii** [comment]
Hey @sampain 
I've opened https://github.com/mui/mui-x/pull/15202 to address this.

**Comment by cherniavskii** [comment]
Closing in favor of https://github.com/mui/mui-x/pull/15202

**Pull request closed without merging**